src/main/java/io/appium/java_client/pagefactory/bys/builder/ByAll.java

```diff
@@ -0,0 +1,52 @@
+package io.appium.java_client.pagefactory.bys.builder;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+
+public class ByAll extends org.openqa.selenium.support.pagefactory.ByAll {
+
+    private final List<By> bys;
+
+    private Function<SearchContext, Optional<WebElement>> getSearchingFunction(By by) {
+        return input -> {
+            try {
+                return Optional.of(input.findElement(by));
+            } catch (NoSuchElementException e) {
+                return Optional.empty();
+            }
+        };
+    }
+
+    /**
+     * @param bys is a set of {@link org.openqa.selenium.By} which forms the all possible searching.
+     */
+    public ByAll(By[] bys) {
+        super(bys);
+        checkNotNull(bys);
+
+        this.bys = Arrays.asList(bys);
+        if (this.bys.isEmpty()) {
+            throw new IllegalArgumentException("By array should not be empty");
+        }
+    }
+
+    @Override
+    public WebElement findElement(SearchContext context) {
+        return bys.stream()
+                .map(by -> getSearchingFunction(by).apply(context))
+                .filter(Optional::isPresent)
+                .findFirst()
+                .orElseThrow(() -> new NoSuchElementException("Cannot locate an element using " + toString()))
+                .orElse(null);
```

Comment:
This is double Optional : `Optional<Optional<WebElement>`. First `orElseThrow` for filter, the second `orElse` for searching function. I can use `get` instead `orElse`, because we already check `Optional::isPresent`, but there is warning  here: "Optional.get()' without 'isPresent()' check"